### DISCUSSION TITLE
Possible conflict between calico ebpf and istio

### DISCUSSION THREAD

**Discussion by zoftdev** [istio/istio#49284] [Q&A]
Hi, I found unexpected result when use istio with calico ebpf dataplane.
Lets' me share my finding.

- Envoy matching rule
	- Find route by Host Header +Port =>  select pod endpoints
	- in case no "Host Header" : Find route by Destination IP+ Port=> select pod endpoints
	
- Environment:
	- Istio 1.20.0  , outboundTrafficPolicy:  REGISTRY_ONLY
	- calico 3.26.4 , enable ebpf dataplance
	
- Situation
	- inside injected: Curl  svcip:svcport -H 'host: svcname.nsname.svc'
	- EBPF detect svcip,svcport then change ip to podip:portport
	- traffic outing to Envoy,
		- envoy find route  by Host Header+port
			- if not match then return 502 / connection reset  Because outboundTrafficPolicy:  REGISTRY_ONLY

I can prove this problem by  if i set servicePort same as podPort.  Traffic will about to route to destination.
So,Are there are solutions to fix this?

**Comment by howardjohn** [comment]
The same bug (in the past) with Cilium: https://github.com/istio/istio/issues/27619

Calico will probably need to fix this.

**Reply by zoftdev** [reply]
Thank God someone found this before :)
